services/users/services.go

```diff
@@ -0,0 +1,50 @@
+package users
+
+import (
+	"errors"
+	"user-crud/models"
+	"user-crud/services"
+	"user-crud/stores"
+)
+
+type userServ struct {
+	u stores.User
+}
+
+func New(st stores.User) services.User {
+	return &userServ{st}
+}
+func (u userServ) Insert(usr models.User) (models.User, error) {
+	usr, err := u.u.Insert(usr)
+	return usr, err
+}
+
+func (u userServ) GetAll() ([]models.User, error) {
+	var users []models.User
+	users, err := u.u.GetAll()
+	return users, err
+}
+
+func (u userServ) GetById(id int) (models.User, error) {
+	usr, err := u.u.GetById(id)
+	return usr, err
+}
+
+func (u userServ) Update(id int, name string) error {
+	err := u.u.Update(id, name)
```

Comment:
Before updating, you should check given id is available or not in the table then go for an update.
also validate id first is id less than 0?